### ISSUE TITLE
Tab completion should be in order of relevance

### ISSUE THREAD

**Issue by Sbozzolo** [jethrokuan/z#97]
I haven't investigated too much, so maybe the answer is obvious. In that case, feel free to immediately close the issue. Consider this example:
```
$ z - l gray
423.664    /home/sbozzolo/documents/projects/2021/gray
143.898    /home/sbozzolo/documents/data/gray/test_fisheye
16         /home/sbozzolo/documents/projects/2021/gray/tests
8          /home/sbozzolo/documents/projects/2021/gray/tools
7.794      /home/sbozzolo/documents/data/gray
[TRUNCATED]
```
With `z gray` I can move me to the first directory, as expected. Sometimes, I want to cd into the second one, so I hit `z gray [TAB]`, resulting in
```
/home/sbozzolo/documents/data/gray                                          /home/sbozzolo/documents/projects/2021/gray-python                  
/home/sbozzolo/documents/data/gray/test_fisheye                             /home/sbozzolo/documents/projects/2021/gray-python/gray             
[TRUNCATED]
```
As you see, the order does not reflect the score, so I have to navigate to the relevant one, which can be multiple key presses away. It would be nice if the order of the suggestions followed the score. This would be in the spirit of `z` and would minimize the number of keystrokes.

**Comment by krobelus** [comment]
that sounds reasonable

**Comment by Sbozzolo** [comment]
Thank you very much!

I updated and it doesn't look like that the problem is solved:
![2021-03-12:13:07:47](https://user-images.githubusercontent.com/9167485/110993052-0b890800-8334-11eb-8352-3d0aab5015ed.png)

The order is different but it doesn't reflect `z -l`

**Comment by krobelus** [comment]
my bad, should be fixed as of 97ca1fd1b281f5f240e7adb90d0a28f9eb4567db
